### DISCUSSION TITLE
What is the current state of IDE (edit: for Scala)?

### DISCUSSION THREAD

**Discussion by tanin47** [playframework/playframework#12281] [General]
Is IntelliJ still the best one? Has anyone tried Eclipse?

Edit: for Scala.

**Comment by jducoeur** [comment]
For Scala or Java development?  (Not obvious, since Play supports both.)

**Reply by tanin47** [reply]
Oops. For Scala. Thank you for asking the clarification question. I'll edit the question.

**Reply by jducoeur** [reply]
In which case, *not* Eclipse.  Once upon a time, the Eclipse-based Scala IDE was the best, but that was long ago -- it hasn't been maintained in many years, and is basically dead.  (AFAIK I was one of the last users, and even I gave up about five years ago.)

At this point, the major options are either IntelliJ or Metals (usually but not necessarily paired with VSC).  They have different strengths, and some folks swear by one or t'other, but both are generally solidly good.

**Comment by Squeng** [comment]
I'm quite happy with the combination of VSC & Metals:
https://scalameta.org/metals/docs/editors/vscode

**Reply by tanin47** [reply]
Wow. I just installed VSCode + Scala Metals. It works well so far. I can't believe the day I move away from IntelliJ would come.
